src/jit/lowerxarch.cpp

```diff
@@ -4573,6 +4579,31 @@ void Lowering::SetMulOpCounts(GenTreePtr tree)
 }
 
 //------------------------------------------------------------------------------
+// SetContainsAVXFlags: Set ContainsAVX flag when it is floating type, set 
+// Contains256bitAVX flag when SIMD vector size is 32 bytes
+//
+// Arguments:
+//    isFloatingType    - true if it is floating type
+//    sizeOfSIMDVector  - SIMD Vector size
+//
+void Lowering::SetContainsAVXFlags(bool isFloatingType /* = true */, unsigned sizeOfSIMDVector /* = 0*/)
+{
+#ifdef FEATURE_AVX_SUPPORT
+    if (comp->getSIMDInstructionSet() == InstructionSet_AVX)
+    {
+        if (isFloatingType)
+        {
+            comp->getEmitter()->SetContainsAVX(true);
+            if (sizeOfSIMDVector == 32)
+            {
+                comp->codeGen->getEmitter()->SetContains256bitAVX(true);
+            }
+        }
+    }
```

Comment:
To be accurate I would suggest to write it as follows

if (isFloatingPointType)
{
    if (comp->getFloatingPointInstructionSet() == InstructionSet_AVX)
     {
           comp->getEmitter()->setContainsAVX(true);
     }

     if (sizeOfSIMDVector == 32 && comp->getSIMDInstructionSet() == InstructionSet_AVX)
     {
           comp->codeGen->getEmitter()->SetContains256bitAVX(true);
     }
}